test/components/admin/datetime_component_test.rb

```diff
@@ -0,0 +1,137 @@
+    assert_equal page.all("select")[1].value, ""
+    assert_equal page.all("label")[2].text, "Year"
+    assert_equal page.all("select")[2].value, ""
+
+    assert_equal page.all("h3")[1].text, "Time"
+    assert_equal page.all("label")[3].text, "Hour"
+    assert_equal page.all("select")[3].value, ""
+    assert_equal page.all("label")[4].text, "Minute"
+    assert_equal page.all("select")[4].value, ""
+  end
+
+  test "populates the datetime fields when a relevant datetime field on the object has a value" do
+    edition = build_stubbed(:edition, first_published_at: Time.zone.local(2020, 1, 1, 12, 30))
+
+    render_inline(
+      Admin::DatetimeComponent.new(
+        object: edition,
+        field_name: "first_published_at",
+      ),
+    )
+
+    assert_equal page.all("select")[0].value, "1"
+    assert_equal page.all("select")[1].value, "1"
+    assert_equal page.all("select")[2].value, "2020"
+    assert_equal page.all("select")[3].value, "12"
+    assert_equal page.all("select")[4].value, "30"
+  end
+
+  test "populates the datetime fields with a default date if one is passed in and the relevant datetime field on the object is nil" do
+    edition = build_stubbed(:edition, first_published_at: nil)
+
+    render_inline(
+      Admin::DatetimeComponent.new(
+        object: edition,
+        field_name: "first_published_at",
+        default_date: Time.zone.local(2022, 2, 2, 11, 45),
+      ),
+    )
+
+    assert_equal page.all("select")[0].value, "2"
+    assert_equal page.all("select")[1].value, "2"
+    assert_equal page.all("select")[2].value, "2022"
+    assert_equal page.all("select")[3].value, "11"
+    assert_equal page.all("select")[4].value, "45"
+  end
+
+  test "prioritises the relevant datetime fields value over a default date" do
+    edition = build_stubbed(:edition, first_published_at: Time.zone.local(2020, 1, 1, 12, 30))
+
+    render_inline(
+      Admin::DatetimeComponent.new(
+        object: edition,
+        field_name: "first_published_at",
+        default_date: Time.zone.local(2022, 2, 2, 11, 45),
+      ),
+    )
+
+    assert_equal page.all("select")[0].value, "1"
+    assert_equal page.all("select")[1].value, "1"
+    assert_equal page.all("select")[2].value, "2020"
+    assert_equal page.all("select")[3].value, "12"
+    assert_equal page.all("select")[4].value, "30"
+  end
+
+  test "uses the start year and and year to populate the year fields options if passed in" do
+    edition = build_stubbed(:edition, first_published_at: Time.zone.local(2020, 1, 1, 12, 30))
+
+    render_inline(
+      Admin::DatetimeComponent.new(
+        object: edition,
+        field_name: "first_published_at",
+        start_year: 1999,
+        end_year: 2022,
+      ),
+    )
+
+    [*1999..2022].each do |year|
+      assert_selector "option[value='#{year}']", count: 1
+    end
+
+    assert_selector "option[value='2023']", count: 0
+    assert_selector "option[value='1998']", count: 0
```

Comment:
```suggestion
    assert_selector "option[value='1999'], option[value='2022']", count: 2
    assert_selector "option[value='1998'], option[value='2023']", count: 0
```

I don't think we need to do the extent of assertions we're doing here as we're just testing the Rails helper, which we already rather trust.